torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -115,7 +115,12 @@ using steady_clock_time_point =
 
 std::chrono::milliseconds getRemainingTime(
     steady_clock_time_point startTime,
-    const std::chrono::milliseconds& timeout) {
+    const std::chrono::milliseconds& timeout,
+    bool waitAllRanks) {
+  if (waitAllRanks) {
+    // See Note in monitoredBarrier
+    return timeout;
```

Comment:
If this is called multiple times, will there be a risk that it can always return a positive remaining time even if the elapsed time is longer than `timeout`?

Reply:
Yes, in the case of `waitAllRanks`, we don't decrease the overall timeout after every call to `wait()`. The reasoning is explained in a comment in `monitoredBarrier`, but essentially if rank n times out, and we want to collect failure info for ranks > n, then the timeout when waiting for ranks > n would always be 0 if a previous rank timed out. Hence we always use the initial timeout for all calls to wait() when `waitAllRanks=true`.